recipes-debian/trousers/trousers_debian.bb

```
@@ -116,3 +116,5 @@ FILES_${PN} += "${systemd_unitdir}/system/tcsd.service"
 CONFFILES_${PN} += "${sysconfig}/tcsd.conf"
 
 BBCLASSEXTEND = "native"
+
+CVE_VERSION = "${@d.getVar('PV').split('+')[0]}"
```

PV should be "0.3.14" in recipes-debian/sources/trousers.inc.
Could you fix the PV instead of this?

---

Is this ok?
```suggestion
PV = "${@d.getVar('PV').split('+')[0]}"
```
or should I fix PV in here?
https://github.com/miraclelinux/meta-debian-extended/blob/1e99e06dbbcd2d352bca1eed7282cf3bfe7f7e59/recipes-debian/sources/trousers.inc#L5

---

Please fix it in sources/trousers.inc.

---

I changed PV and REPACK_PV to "0.3.14" in recipes-debian/sources/trousers.inc.
then, "bitbake trousers" command got following error in do_debian_patch.
```
$ bitbake trousers
...
ERROR: trousers-0.3.14-r0 do_debian_patch: Function failed: do_debian_patch (log file is located at /home/ubuntu/emlinux-2.1-202006/build/tmp-glibc/work/aarch64-emlinux-linux/trousers/0.3.14-r0/temp/log.do_debian_patch.2066)
ERROR: Logfile of failure stored in: /home/ubuntu/emlinux-2.1-202006/build/tmp-glibc/work/aarch64-emlinux-linux/trousers/0.3.14-r0/temp/log.do_debian_patch.2066
Log data follows:
| DEBUG: Executing python function extend_recipe_sysroot
| NOTE: Direct dependencies are ['/home/ubuntu/emlinux-2.1-202006/repos/meta-debian/recipes-debian/quilt/quilt-native_debian.bb:do_populate_sysroot']
| NOTE: Installed into sysroot: []
| NOTE: Skipping as already exists in sysroot: ['quilt-native']
| DEBUG: Python function extend_recipe_sysroot finished
| DEBUG: SITE files ['endian-little', 'bit-64', 'arm-common', 'arm-64', 'common-linux', 'common-glibc', 'aarch64-linux', 'common']
| DEBUG: Executing shell function do_debian_patch
| NOTE: Debian source format is '3.0 (quilt)'
| Applying patch 01-tss-user.patch
| can't find file to patch at input line 12
| Perhaps you used the wrong -p or --strip option?
| The text leading up to this was:
| --------------------------
| |Author: Daniel Baumann <daniel@debian.org>
| |Author: Pierre Chifflier <pollux@debian.org>
| |Description: Avoiding chown/chmod for tss user at build-time.
| | Note that we do not use NOUSERCHECK or --enable-usercheck=no,
| | since these are also used in the sources and will change the
| | binary.
| |
| |Index: trousers/dist/Makefile.am
| |===================================================================
| |--- trousers.orig/dist/Makefile.am
| |+++ trousers/dist/Makefile.am
| --------------------------
| No file to patch.  Skipping patch.
| 1 out of 1 hunk ignored
| Patch 01-tss-user.patch does not apply (enforce with -f)
| WARNING: exit code 1 from a shell command.
| ERROR: Function failed: do_debian_patch (log file is located at /home/ubuntu/emlinux-2.1-202006/build/tmp-glibc/work/aarch64-emlinux-linux/trousers/0.3.14-r0/temp/log.do_debian_patch.2066)
ERROR: Task (/home/ubuntu/emlinux-2.1-202006/repos/meta-debian-extended/recipes-debian/trousers/trousers_debian.bb:do_debian_patch) failed with exit code '1'
...
```

To resolve this, I added  DEBIAN_UNPACK_DIR in recipes-debian/trousers/trousers_debian.bb
```
DEBIAN_UNPACK_DIR = "${WORKDIR}/trousers-0.3.14+fixed1"
```

After this fix, I confirmed that "bitbake trousers" completed successfully.
If it looks okay, I re-push commits.

---

Could you use `"${WORKDIR}/${BPN}-${REPACK_PV}"` to reduce future changes?

https://github.com/meta-debian/meta-debian/blob/644f693240fc55fa15fdc5e11a25510961c7684f/recipes-debian/bison/bison_debian.bb#L22

---

Thank you for your advice.
I fix in that way and re-push.

---

I'm sorry, we should not modify sources/*inc files because they are automatically generated. So your first change was correct. I made a PR for that.
https://github.com/miraclelinux/meta-debian-extended/pull/275